modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/NewCopyWithDefaultVersion.java

```diff
@@ -59,107 +57,85 @@
-
-        //add api
-        HttpResponse apiAddRequest = apiPublisher.addAPI(apiCreationRequestBean);
-        assertEquals(apiAddRequest.getResponseCode(), Response.Status.OK.getStatusCode(),
+        //API request
+        APIRequest apiRequest = new APIRequest(API_NAME, apiContext, new URL(endpointUrl));
+        apiRequest.setTags(TAGS);
+        apiRequest.setDescription(description);
+        apiRequest.setVersion(version);
+        apiRequest.setProvider(providerName);
+        apiRequest.setEndpointType(endPointType);
+        apiRequest.setResourceMethodAuthType(resourceMethodAuthType);
+        apiRequest.setTier(tier);
+        apiRequest.setResourceMethodThrottlingTier(resTier);
+        apiRequest.setUriTemplate(uriTemplate);
+        apiRequest.setVisibility(visibility);
+
+        //Add API
+        HttpResponse serviceResponse = restAPIPublisher.addAPI(apiRequest);
+        apiId = serviceResponse.getData();
+        assertEquals(serviceResponse.getResponseCode(), Response.Status.CREATED.getStatusCode(),
                 "Invalid Response Code");
-        assertTrue(apiAddRequest.getData().contains("{\"error\" : false}"),
-                "Response Data Mismatched");
 
         //copy api with default version
-        HttpResponse apiCopyResponse = apiPublisher.copyAPI(providerName,API_NAME,version,newVersion,
-                "default_version");
-        assertEquals(apiCopyResponse.getResponseCode(),Response.Status.OK.getStatusCode(),
+        HttpResponse apiCopyResponse = restAPIPublisher.copyAPI(newVersion, apiId, true);
+        defaultApiId = apiCopyResponse.getData();
+        assertEquals(apiCopyResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
                 "Response Code Mismatch");
-        assertTrue(apiCopyResponse.getData().contains("{\"error\" : false}"),
-                "Response Data Mismatched");
-
-        HttpResponse newVersionApi = apiPublisher.getAPI(API_NAME,providerName,newVersion);
-        assertEquals(newVersionApi.getResponseCode(),Response.Status.OK.getStatusCode(),
-                "Response Code Mismatched");
-
-        JSONObject jsonObject = new JSONObject(newVersionApi.getData());
-
-        String name = jsonObject.getJSONObject("api").getString("name");
-        assertEquals(name,API_NAME,"API name is mismatched");
 
-        String version = jsonObject.getJSONObject("api").getString("version");
-        assertEquals(version,newVersion,"API Version is mismatched");
-
-        boolean defaultVersion = jsonObject.getJSONObject("api").getBoolean("isDefaultVersion");
-        assertEquals(defaultVersion,true,"Error in Default Version selection");
-
-        String currentDefaultVersionCheck = jsonObject.getJSONObject("api").getString("currentDefaultVersion");
-        assertEquals(currentDefaultVersionCheck,newVersion,"Error in Default Version");
+        HttpResponse newVersionApi = restAPIPublisher.getAPI(defaultApiId);
+        Gson g = new Gson();
```

Comment:
shall we use a more meaningful name like 'gson' rather than 'g'?

Reply:
fixed